### PR TITLE
[fix](compaction) block reader should check column string offset

### PR DESCRIPTION
**Pull request by hust-hhb**
## Proposed changes
there is some bug like cats will make data wrong, compaction will core when reading wrong data, so need to add a check on column string.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hust-hhb** [comment]
run buildall

File: be/src/vec/olap/vertical_block_reader.cpp [hunk 1]
```diff
@@ -524,10 +524,10 @@ Status VerticalBlockReader::_unique_key_next_block(Block* block, bool* eof) {
         }
         const auto& src_block = _next_row.block;
         assert(src_block->columns() == column_count);
-        for (size_t i = 0; i < column_count; ++i) {
+        RETURN_IF_CATCH_EXCEPTION(for (size_t i = 0; i < column_count; ++i) {
```
Comment (yiguolei) [inline be/src/vec/olap/vertical_block_reader.cpp:527]: not catch the loop.
just catch the internal  insert_from is ok

File: be/src/vec/olap/block_reader.cpp [hunk 2]
```diff
@@ -407,12 +407,14 @@ Status BlockReader::_unique_key_next_block(Block* block, bool* eof) {
     return Status::OK();
 }
 
-void BlockReader::_insert_data_normal(MutableColumns& columns) {
+Status BlockReader::_insert_data_normal(MutableColumns& columns) {
     auto block = _next_row.block.get();
-    for (auto idx : _normal_columns_idx) {
+    RETURN_IF_CATCH_EXCEPTION(for (auto idx
```
Comment (yiguolei) [inline be/src/vec/olap/block_reader.cpp:412]: not catch the for loop
catch the insert from method

File: be/src/vec/olap/block_reader.h [hunk 3]
```diff
@@ -80,7 +80,7 @@ class BlockReader final : public TabletReader {
 
     void _init_agg_state(const ReaderParams& read_params);
 
-    void _insert_data_normal(MutableColumns& columns);
+    Status _insert_data_normal(MutableColumns& columns);
```
Comment (yiguolei) [inline be/src/vec/olap/block_reader.h:83]: add [no discard] attribute

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Pull request closed without merging**